src/tlo/methods/symptommanager.py

```diff
@@ -460,33 +465,81 @@ def who_not_have(self, symptom_string: str) -> pd.Index:
             )
         ]
 
-    def has_what(self, person_id, disease_module: Module = None):
+    def has_what(
+        self,
+        person_id: Optional[int] = None,
+        individual_details: Optional[IndividualProperties] = None,
+        disease_module: Optional[Module] = None,
+    ) -> List[str]:
         """
         This is a helper function that will give a list of strings for the symptoms that a _single_ person
         is currently experiencing.
-        Optionally can specify disease_module_name to limit to the symptoms caused by that disease module
 
-        :param person_id: the person_of of interest
-        :param disease_module: (optional) disease module of interest
-        :return: list of strings for the symptoms that are currently being experienced
-        """
+        If working in a `tlo.population.IndividualProperties` context, one can pass the context object
+        instead of supplying the person's DataFrame index.
+        Note that at least one of these inputs must be passed as a keyword argument however.
+        In the event that both arguments are passed, the individual_details argument takes precedence over the person_id.
 
-        assert isinstance(person_id, (int, np.integer)), 'person_id must be a single integer for one particular person'
+        Optionally can specify disease_module_name to limit to the symptoms caused by that disease module.
 
-        df = self.sim.population.props
-        assert df.at[person_id, 'is_alive'], "The person is not alive"
-
-        if disease_module is not None:
-            assert disease_module.name in ([self.name] + self.recognised_module_names), \
-                "Disease Module Name is not recognised"
-            sy_columns = [self.get_column_name_for_symptom(s) for s in self.symptom_names]
-            person_has = self.bsh.has(
-                [person_id], disease_module.name, first=True, columns=sy_columns
-            )
-            return [s for s in self.symptom_names if person_has[f'sy_{s}']]
+        :param person_id: the person_of of interest.
+        :param individual_details: `tlo.population.IndividualProperties` object for the person of interest.
+        :param disease_module: (optional) disease module of interest.
+        :return: list of strings for the symptoms that are currently being experienced.
+        """
+        assert (
+            disease_module.name in ([self.name] + self.recognised_module_names)
+            if disease_module is not None
+            else True
+        ), "Disease Module Name is not recognised"
+
+        if individual_details is not None:
+            # We are working in an IndividualDetails context, avoid lookups to the
+            # population DataFrame as we have this context stored already.
+            assert individual_details["is_alive"], "The person is not alive"
```

Comment:
I know this replicates the existing behaviour but this check probably belongs somewhere else. It's conceivable to want to know the symptoms of deceased individual. @tbhallett ?

Reply:
I'm not sure if we would need to know the symptoms of someone who has already died. We may wish to log symptoms of someone at the point of death (like we log other characteristics) but that's different.
I think throughout the code we assume that the information from deceased individuals is deleted (even though its not)